### PR TITLE
fix Image name to match new requirement

### LINKED ISSUE #185: Image name need to be changed
now, the image name needs to be like "docker-address/userName/pipelineName: timestamp-commit_id" as a default image name if the user does not input image name.

### PR DESCRIPTION
**Pull request by zz-bmj**
## Fixes #185

### CODE REVIEW

File: worker/worker.go [hunk 1]
```diff
@@ -186,9 +186,23 @@ func (worker *Worker) createVersion(vcsManager *vcs.Manager, event *api.Event) {
...
-	event.Data["image-name"] = fmt.Sprintf("%s/%s/%s", dockerManager.Registry,
-		strings.ToLower(event.Service.Username), strings.ToLower(event.Service.Name))
-	event.Data["tag-name"] = event.Version.Name
+
+	var imageName, tagName string
+
+	if "" == event.Service.ImageName {
+		imageName = fmt.Sprintf("%s/%s/%s", dockerManager.Registry,
+			strings.ToLower(event.Service.Username), strings.ToLower(event.Service.Name))
+		tagName = fmt.Sprintf("%d-%s", time.Now().Unix(), event.Version.Commit)
```
Comment (supereagle) [inline worker/worker.go]: What about if `event.Version.Commit` is empty? 
The format should be `{short commit}-{timestamp}`, like `7654321-170401040120`.

File: worker/worker.go [hunk 2]
```diff
@@ -186,9 +186,23 @@ func (worker *Worker) createVersion(vcsManager *vcs.Manager, event *api.Event) {
...
+	} else {
+		if arr := strings.Split(event.Service.ImageName, ":"); len(arr) == 2 {
+			imageName, tagName = arr[0], arr[1]
+		} else {
+			setEventFailStatus(event, err.Error())
```
Comment (supereagle) [inline worker/worker.go]: Here the `err` is nil, must new an error to illustrate the fail reason.

File: worker/worker.go (hunk 1 above)
Comment (zz-bmj) [reply worker/worker.go]: done

File: worker/worker.go (hunk 2 above)
Comment (zz-bmj) [reply worker/worker.go]: done

**Comment by zz-bmj** [comment]
@supereagle PTAL

**Comment by supereagle** [comment]
LGTM

**Pull request merged**